ui_advanced_config.py

```diff
@@ -617,291 +632,509 @@ def updateComponentsEntryWidgets(self, v):
+        self.currentMapData = zmap.Map(
+            {
+                "name": "",
+                "edge_obj_id": "",
+                "desc": "",
+                "width": "",
+                "height": "",
+                "placed_objects": {},
+                "placed_items": {},
+                "sides": {},
+                "win_states": [],
+            }
+        )
+        self.showMapEntry(self.currentMapData)
+
+    ### DELETE ###
+
+    def deleteObject(self):
+        if self.selectObjectsCombo.get() in self.objectData:
+            self.objectData.pop(self.selectObjectsCombo.get())
+            self.objectIDs.pop(self.selectObjectsCombo.current())
+            self.selectObjectsCombo.configure(values=self.objectIDs)
+            self.selectObjectsCombo.current(len(self.objectIDs) - 1)
+            self.changeObjectsEntryWidgets()
+
+    def deleteTeam(self):
+        if self.selectTeamCombo.get() in self.teamData:
+            self.teamData.pop(self.selectTeamCombo.get())
+            self.teamNames.pop(self.selectTeamCombo.current())
+
+            with open("settings/teams.json", "r") as f:
+                teamJSON = json.load(f)
+            f.close()
+            teamJSON.pop(self.selectTeamCombo.get())
+            with open("settings/teams.json", "w") as f:
+                json.dump(teamJSON, f, indent=4)
+            f.close()
+
+            self.selectTeamCombo.configure(values=self.teamNames)
+            self.selectTeamCombo.current(len(self.teamNames) - 1)
+            self.changeTeamEntryWidgets()
+
+    def deleteComponents(self):
+        if self.selectComponentCombo.get() in self.componentData:
+            self.componentData.pop(self.selectComponentCombo.get())
+
+            with open("settings/components.json", "r") as f:
+                componentJSON = json.load(f)
+                componentJSON.pop(self.selectComponentCombo.get())
+            f.close()
+            with open("settings/components.json", "w") as f:
+                json.dump(componentJSON, f, indent=4)
+            f.close()
+        self.componentIDs.pop(self.selectComponentCombo.current())
+        self.selectComponentCombo.configure(values=self.componentIDs)
+        self.selectComponentCombo.current(len(self.componentIDs) - 1)
+        self.changeComponentsEntryWidgets()
+
+    def deleteMap(self):
+        if self.selectMapsCombo.get() in self.mapData:
+            self.mapData.pop(self.selectMapsCombo.get())
+            with open("settings/maps.json", "r") as f:
+                mapJSON = json.load(f)
+                mapJSON.pop(self.selectMapsCombo.get())
+            f.close()
+            with open("settings/maps.json", "w") as f:
+                json.dump(mapJSON, f, indent=4)
+            f.close()
+        self.mapIDs.pop(self.selectMapsCombo.current())
+        self.selectMapsCombo.configure(values=self.mapIDs)
+        self.selectMapsCombo.current(len(self.mapIDs) - 1)
+        self.changeMapsEntryWidgets()
```

Comment:
Overall amazing work and massive amount of accomplishment here, with some bugs I have noted through testing of config buttons (may be additional I have missed). Recommend we take a deep dive during next code review on this PR to try to find any additional bugs.